Pcap++/header/PcapLiveDevice.h

```diff
@@ -35,7 +36,7 @@ namespace pcpp
 	 * @param[in] pDevice A pointer to the PcapLiveDevice instance
 	 * @param[in] userCookie A pointer to the object put by the user when packet capturing stared
 	 */
-	typedef void (*OnPacketArrivesCallback)(RawPacket* pPacket, PcapLiveDevice* pDevice, void* userCookie);
+	using OnPacketArrivesCallback = std::function<void(RawPacket*, PcapLiveDevice*, void*)>;
```

Comment:
I think we should modify the doxygen documentation. The `@typedef` keyword is no longer relevant. I'm not sure how to document a `using` statement, and also how to document its parameters 🤔 

This also applies to the other changes below

Reply:
`using` is sometimes used for namespace, however here is for the type alias.

There is no specific notation for `using` but I think using `@typedef` is fine here, because the overall idea is the same.

I just checked the doxygen output, I think it's good to keep the current doxygen comment.
![image](https://github.com/seladb/PcapPlusPlus/assets/18013815/8a8a7333-54b6-479f-9d84-eb2d29aba305)

Reply:
another choice here is to use `@var`, but I don't think there is difference.

Reply:
I tried to remove the `@typedef` lines and it still works, so maybe we can do that?

Also - please notice I changed `pPacket` to `packet` and `pDevice` to `device`:

```cpp
	/**
	 * A callback that is called when a packet is captured by PcapLiveDevice
	 * @param[in] packet A pointer to the raw packet
	 * @param[in] device A pointer to the PcapLiveDevice instance
	 * @param[in] userCookie A pointer to the object put by the user when packet capturing stared
	 */
	using OnPacketArrivesCallback = std::function<void(RawPacket*, PcapLiveDevice*, void*)>;

	/**
	 * A callback that is called when a packet is captured by PcapLiveDevice
	 * @param[in] packet A pointer to the raw packet
	 * @param[in] device A pointer to the PcapLiveDevice instance
	 * @param[in] userCookie A pointer to the object put by the user when packet capturing stared
	 * @return True when main thread should stop blocking or false otherwise
	 */
	using OnPacketArrivesStopBlocking = std::function<bool(RawPacket*, PcapLiveDevice*, void*)>;


	/**
	 * A callback that is called periodically for stats collection if user asked to start packet capturing with periodic stats collection
	 * @param[in] stats A reference to the most updated stats
	 * @param[in] userCookie A pointer to the object put by the user when packet capturing stared
	 */
	using OnStatsUpdateCallback = std::function<void(IPcapDevice::PcapStats&, void*)>;
```

![image](https://github.com/seladb/PcapPlusPlus/assets/9059541/79a1bab8-907d-4f7e-a1f9-afa9eaacfc33)

Reply:
addressed